app/controllers/users/reset_passwords_controller.rb

```diff
@@ -21,19 +22,12 @@ def create
     end
 
     def edit
-      result = PasswordResetTokenValidator.new(token_user).submit
-
-      analytics.password_reset_token(**result.to_h)
-      irs_attempts_api_tracker.forgot_password_email_confirmed(
-        success: result.success?,
-        failure_reason: irs_attempts_api_tracker.parse_failure_reason(result),
-      )
-
-      if result.success?
+      if params[:reset_password_token] && FeatureManagement.redirect_to_clean_edit_password_url?
+        redirect_to edit_user_password_url
+      else
         @reset_password_form = ResetPasswordForm.new(build_user)
         @forbidden_passwords = forbidden_passwords(token_user.email_addresses)
-      else
-        handle_invalid_or_expired_token(result)
+        session.delete(:reset_password_token)
```

Comment:
If I refresh the page when changing my password, I see an error "The reset password token is invalid". I think we should support the user reloading the page.

Should we wait to delete the session token until after the submission? Or is this meant to "undo" what was stored in the session at line 90 if the feature is disabled? IMO, we should only assign into the session when we intend it to be used, rather than always assigning and then immediately unassigning in the scenarios where we don't want it.

![image](https://user-images.githubusercontent.com/1779930/234274197-c19dd7c4-735d-430f-8c08-56199e0a0136.png)